src/compiler/types.ts

```diff
@@ -1985,8 +1990,7 @@ export interface ShorthandPropertyAssignment extends ObjectLiteralElement, JSDoc
     readonly objectAssignmentInitializer?: Expression;
 
     // The following properties are used only to report grammar errors
-    /** @internal */ readonly illegalDecorators?: NodeArray<Decorator> | undefined; // shorthand property assignment cannot have decorators
-    /** @internal */ readonly modifiers?: NodeArray<Modifier> | undefined; // shorthand property assignment cannot have modifiers
+    /** @internal */ readonly decoratorsAndModifiers?: NodeArray<ModifierLike> | undefined; // shorthand property assignment cannot have decorators or modifiers

```

Comment:
As we already discussed, I really think we should just have a single `modifiers` bucket for everything. The distinction between these two buckets is just way too confusing.